src/groupeddataframe/groupeddataframe.jl

```diff
@@ -387,6 +391,29 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::AbstractDict{<:AbstractString} , gd)
+    T = keytype(key)
```

Comment:
Also check if `key` does not contain other entries than these in `gd.cols`. It is enough to test length, as other mismatches will be captured by indexing.

Reply:
ah - you check it below, but I think it would be better to check it in these methods and drop a check in the calling methods (I feel it would be less error prone in the long run).